preprocessing/processing.py

```diff
@@ -0,0 +1,272 @@
+"""
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+
+import os
+import sys
+from collections import Counter, defaultdict
+import json
+import numpy as np
+import pandas as pd
+
+cwd = os.getcwd()
+sys.path.append(os.path.join(cwd, "preprocessing"))
+sys.path.append(os.path.join(cwd, 'preprocessing/raw_data/top-dataset-semantic-parsing'))
+
+from tree import Tree, Intent, Slot
+
+raw_data_dir = os.path.join(cwd, 'preprocessing/raw_data')
+data_file = ['train', 'test', 'dev']
+
+
+# preprocess ATIS data
+atis_dir = 'atis_resplit'
+atis_out = os.path.join(cwd, 'data/atis')
+intent_vocab = defaultdict(set)
+slot_vocab = defaultdict(set)
+
+for file in data_file:
+
+    atis_df = pd.read_csv(os.path.join(raw_data_dir, atis_dir, f'atis.{file}.csv'))
+
+    token_list = atis_df.tokens.apply(lambda s: s.split(' '))
+    tag_list = atis_df.slots.apply(lambda s: s.split(' '))
+
+    data_points = []
+    for i in range(len(atis_df)):
+        intents = atis_df.intent[i].split("#")
+        intent_vocab[file].update(set(intents))
+        slot_vocab[file].update(set(np.unique(tag_list[i])))
+
+        id = atis_df.id[i]
+        token_list[i][0] = "<BOS>"
+        token_list[i][-1] = "<EOS>"
+        token_tag_pair = [f"{token} {tag}" for token, tag in zip(token_list[i], tag_list[i])]
+
+        for j in range(1, len(token_list[i])):
```

Comment:
For readability, it may help to add a comment here as to what is being done in this loop.